src/test/java/org/apache/commons/compress/compressors/DetectCompressorTestCase.java

```diff
@@ -167,6 +171,26 @@ private String detect(String testFileName) throws IOException, CompressorExcepti
     }
 
     @Test
+    public void testMemoryLimit() throws Exception {
+        testMemoryLimit("COMPRESS-382");
+        testMemoryLimit("COMPRESS-386");
+    }
+
+    private void testMemoryLimit(String fileName) throws IOException, CompressorException {
+        CompressorStreamFactory fac = new CompressorStreamFactory(true,
+                100);
+        try (InputStream is = new BufferedInputStream(
+                new FileInputStream(getFile(fileName)))) {
+            InputStream compressorInputStream = fac.createCompressorInputStream(is);
+            fail("Should have thrown CompressorMemoryLimitException");
```

Comment:
Maybe instead we could let the CompressorMemoryLimitException happen, and declare in the @Test annotation the expected exception.